### ISSUE TITLE
Feature request: Support multidimensional `batch_dim`

### ISSUE THREAD

**Issue by ragulpr** [ragulpr/taildropout#1]
Currently, each dimension except the `batch_dim` will share the same mask. This might be unexpected behaviour, example that each sequence would have the same mask. 

Current behaviour:
```
TailDropout(p=0.5, batch_dim=0, dropout_dim = -1)(torch.ones(3,2,10)) # <- default
tensor([[[1., 1., 1., 1., 1., 1., 1., 1., 0., 0.],
         [1., 1., 1., 1., 1., 1., 1., 1., 0., 0.]],

        [[1., 1., 1., 0., 0., 0., 0., 0., 0., 0.],
         [1., 1., 1., 0., 0., 0., 0., 0., 0., 0.]],

        [[0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
         [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]]])
```
Would be nice to be able to call something like;
``TailDropout(p=0.5, batch_dim=(0,1,), dropout_dim = -1)(torch.ones(3,2,10)) ``
